sysid-application/src/main/native/cpp/view/Generator.cpp

```diff
@@ -516,7 +516,21 @@ void Generator::Display() {
       "Encoders (REV already handles this value): 1");
 
   // Add gearing
-  ImGui::InputDouble("Gearing", &m_settings.gearing, 0.0, 0.0, "%.2f");
+  ImGui::SetNextItemWidth(ImGui::GetFontSize() * 4);
+  ImGui::InputDouble("##1", &m_settings.gearingNumerator, 0.0, 0.0, "%.2f");
+  ImGui::SameLine();
+  ImGui::Text(":");
+  ImGui::SameLine();
+  ImGui::SetNextItemWidth(ImGui::GetFontSize() * 4);
+  ImGui::InputDouble("##2", &m_settings.gearingDenominator, 0.0, 0.0, "%.2f");
+  ImGui::SameLine();
+  ImGui::Text("Gearing");
+
+  // Ensure no division by zero
+  if (m_settings.gearingDenominator <= 0) {
+    m_gearingDenominator = 1;
+  }
```

Comment:
We should do this check for the numerator as well since a zero or negative value won't do sane things either.